firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -126,101 +127,133 @@ String getName() {
   }
 
   /**
-   * Create firebase installation id for the {@link FirebaseApp} on FIS Servers and client side
-   * cache.
+   * Get firebase installation id for the {@link FirebaseApp}. Either from the persisted fid entry
+   * if already exists and is not error state or create a new Fid on the FIS Servers and persist in
+   * the shared prefs.
    *
    * <pre>
    *     The workflow is:
-   *         check if cache empty or cache status is REGISTER_ERROR
+   *          Persisted Fid does not exists Or Persisted Fid is in Error state
    *                                 |
-   *                           create random fid
-   *                                 |
-   *               update cache with cache status UNREGISTERED
-   *                                 |
-   *                           send http request to backend
-   *                                 |                  |
-   *  on success: set cache entry status to REGISTERED  |
-   *                                 |                  |
-   *                                 |         on failure: set cache entry status to REGISTER_ERROR
-   *                                 |                     |
-   *                                return          throw exception
+   *             return a newly created random Fid and persist the Fid entry
+   *
+   *
+   *
+   *                    Persisted Fid exists And is Registered
+   *                                  |
+   *                          return the Persisted Fid
+   *
+   *
    *
+   *                  Persisted Fid exists and is in Unregistered state
+   *                                     |
+   *                Register the Fid on FIS backened and update the PersistedFidEntry
```

Comment:
Please replace "backened" with "server" (or "API").

Reply:
I see this a lot at Google, but I will continue preaching:
My take on the term "backend":
The term "backend" only really makes sense in the context of "frontend vs. backend".
Looking at our FIS API, the OnePlatform Service is the frontend (that accepts requests). Our Spanner database is the backend (that stores data).
Looking at our FIS SDK, there is not really a frontend or backend, but if anything, the public methods are the frontend (that can be called by other SDKs or the App), the local storage is the backend (where FIDs and tokens are persisted).
I think in this case the terminology is rather client (our SDK) vs. server (our API).
Thus, I recommend to either use the term "server" or "API".

Reply:
Done.